homeassistant/helpers/entity.py

```diff
@@ -314,21 +324,37 @@ def has_entity_name(self) -> bool:
             return self.entity_description.has_entity_name
         return False
 
+    def _device_class_name(self) -> str | None:
+        """Return a translated name of the entity based on its device class."""
+        assert self.platform
+        if not self.has_entity_name:
+            return None
+        device_class_key = self.device_class or "_"
+        name_translation_key = (
+            f"component.{self.platform.domain}.entity_component."
+            f"{device_class_key}.name"
+        )
+        return self.platform.component_translations.get(name_translation_key)
+
     @property
     def name(self) -> str | None:
         """Return the name of the entity."""
         if hasattr(self, "_attr_name"):
+            if self._attr_name is DEVICE_CLASS_NAME:
```

Comment:
What is the plan to make this come true? If it's not the default, this change will not help much.

Reply:
An integration needs to set either `Entity._attr_name` or `EntityDescription.name` to `DEVICE_CLASS_NAME`.
This is less work than setting a `translation_key` + updating `strings.json` with a reference to the entity components' translation.

I don't think letting either `Entity._attr_name` or `EntityDescription.name` default to `DEVICE_CLASS_NAME` is possible.